addons/stock/models/stock_move.py

```diff
@@ -523,14 +523,19 @@ def _set_lot_ids(self):
                     move_line.qty_done = 1
             move.write({'move_line_ids': move_lines_commands})
 
-    @api.depends('picking_type_id', 'date', 'priority')
+    @api.depends('picking_type_id', 'picking_type_id.reservation_method', 'picking_type_id.reservation_days_before', 'picking_type_id.reservation_days_before_priority', 'date', 'state', 'priority')
```

Comment:
Hmm, it is, but as it is quite specific in the inside and really depending on the picking type that things are not working out...this was the reason why I came here by the way, so I do not agree as I think we need the consistency otherwise the feature is just crap and non-deterministic (sometimes working, sometimes not and the user has no option to know why)

Reply:
It is better to add a warning (during an on_change of reservation method), or add an action server to recompute.
This kind of big trigger is very bad (after a migration for exemple, when user want update picking type).

Reply:
But is it really that big? Even if you have millions of done moves they would be all `False` already and therefore unchanged whereas the only changes are taking place for certain moves which are subject to reservation and I think if the field is not dirty after the "quick" python iteration it will not even trigger a write to the database for those unchanged moves (at least I hope we are already there) and ideally it is just a single query which I would expect to finish the computation.

Anyway, I got your point, but consistency is, at least for me, much more important than some fear of the unavoidable...

About the action, yes this can be always done, but needs someone like us to trigger or do a scheduler task for that, which might be a solution, but still I think we have to stop to fix the system by our actions...it needs to work for itself consistently...whatever it takes, but yeah, probably no one will merge this anyway as first a dozen of customers have to complain to take a risk where even the both of us are not of the same opinion.

I really hope at some point we get even better tools and highly deterministic tests and implementations to see it right away what it would cost 😉

Reply:
Also we could mitigate prefetching which might be greedy to only touch the minimum we need to do a recordset as big as you want to process. Anyway, I keep it in my fork commits and will immediately see in bigger databases what needs to be tweaked at some point.